src/main/java/com/box/sdk/BoxSearchParameters.java

```diff
@@ -0,0 +1,427 @@
+    }
+    /**
+     * Set the file size range upper bounds Bytes.
+     * @param sizeRangeUpperBoundBytes a size filter that will be upper range.
+     */
+    public void setSizeRangeUpperBoundBytes(int sizeRangeUpperBoundBytes) {
+        this.sizeRangeUpperBoundBytes = sizeRangeUpperBoundBytes;
+    }
+    /**
+     * Return the list of owner id's that are being applied as a search filter on the results.
+     * @return ownerUserIds.
+     */
+    public List<String> getOwnerUserIds() {
+        return this.ownerUserIds;
+    }
+    /**
+     * Set the owner id's to be applied as a filter to restrict the results on specific file owners.
+     * @param ownerUserIds applied ownerId's.
+     */
+    public void setOwnerUserIds(List<String> ownerUserIds) {
+        this.ownerUserIds = ownerUserIds;
+    }
+    /**
+     * Return the list of folder ids that is currently being used as applied filter.
+     * @return  ancestorFolderIds.
+     */
+    public List<String> getAncestorFolderIds() {
+        return this.ancestorFolderIds;
+    }
+    /**
+     * Set the list of folder id's to be applied as a filter on the search filters.
+     * @param ancestorFolderIds a list of folder ids that will contain results to specific folders.
+     */
+    public void setAncestorFolderIds(List<String> ancestorFolderIds) {
+        this.ancestorFolderIds = ancestorFolderIds;
+    }
+    /**
+     * Return content types that or being applied as a filter (name,description,tags,file_content).
+     * @return  contentTypes.
+     */
+    public List<String> getContentTypes() {
+        return this.contentTypes;
+    }
+    /**
+     * Set list of content types that will be used as a filters.
+     * @param contentTypes a list of content types such as (name,description,tags,file_content).
+     */
+    public void setContentTypes(List<String> contentTypes) {
+        this.contentTypes = contentTypes;
+    }
+    /**
+     * Return the type you want to return in your search.
+     * @return String type.
+     */
+    public String getType() {
+        return this.type;
+    }
+    /**
+     * Set the type you want to search for can be file, folder, or web_link.
+     * @param type can be file, folder, or web_link.
+     */
+    public void setType(String type) {
+        this.type = type;
+    }
+    /**
+     * Return the specified trash search preference.
+     * @return String trashContent.
+     */
+    public String getTrashContent() {
+        return this.trashContent;
+    }
+    /**
+     * Set trash filter. Can be trashed_only or non_trashed_only, without this parameter default to non_trashed_only.
+     * @param trashContent Can be trashed_only or non_trashed_only.
+     */
+    public void setTrashContent(String trashContent) {
+        this.trashContent = trashContent;
+    }
+    /**
+     * Retrieve the existing BoxMetaDataFilterRequest.
+     * @return this.BoxMetaDataFilterRequest
+     */
+    public BoxMetadataFilterRequest getMdFilters() {
```

Comment:
What do you think of having this return a List of BoxMetadataFilter objects instead?